modules/geo-layers/src/tile-layer/utils.js

```diff
@@ -42,18 +42,37 @@ export function getURLFromTemplate(template, properties) {
 /**
  * gets the bounding box of a viewport
  */
-function getBoundingBox(viewport) {
-  const corners = [
-    viewport.unproject([0, 0]),
-    viewport.unproject([viewport.width, 0]),
-    viewport.unproject([0, viewport.height]),
-    viewport.unproject([viewport.width, viewport.height])
-  ];
+function getBoundingBox(viewport, zRange) {
+  let corners;
+  if (zRange && zRange.length === 2) {
+    const [minZ, maxZ] = zRange;
+    corners = [
+      // Lower zRange
+      viewport.unproject([0, 0], {targetZ: minZ}),
```

Comment:
Nit: define `{targetZ: minZ}` and `{targetZ: maxZ}` as variables to avoid re-creating small objects.